Scripts/RemediationScripts/Remediate-DisableNonSSLPortOnRedisCache.ps1

```diff
@@ -0,0 +1,680 @@
+        Rolls back remediation done for 'Azure_RedisCache_DP_Use_SSL_Port' Control.
+        Enable Non-SSL port on Redis Cache(s) in the Subscription. 
+        
+        .PARAMETER SubscriptionId
+        Specifies the ID of the Subscription that was previously remediated.
+        
+        .PARAMETER Force
+        Specifies a forceful roll back without any prompts.
+        
+        .Parameter PerformPreReqCheck
+        Specifies validation of prerequisites for the command.
+      
+        .PARAMETER FilePath
+        Specifies the path to the file to be used as input for the roll back.
+
+        .INPUTS
+        None. You cannot pipe objects to Enable-NonSSLPortOnRedisCache.
+
+        .OUTPUTS
+        None. Enable-NonSSLPortOnRedisCache does not return anything that can be piped and used as an input to another command.
+
+        .EXAMPLE
+        PS> Enable-NonSSLPortOnRedisCache -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\EnableSecurityScanningIdentityForRedisCache\RemediatedRedisCache.csv
+
+        .LINK
+        None
+    #>
+
+    param (
+        [String]
+        [Parameter(Mandatory = $true, HelpMessage="Specifies the ID of the Subscription that was previously remediated.")]
+        $SubscriptionId,
+
+        [Switch]
+        [Parameter(HelpMessage="Specifies a forceful roll back without any prompts")]
+        $Force,
+
+        [Switch]
+        [Parameter(HelpMessage="Specifies validation of prerequisites for the command")]
+        $PerformPreReqCheck,
+
+        [String]
+        [Parameter(Mandatory = $true, HelpMessage="Specifies the path to the file to be used as input for the roll back")]
+        $FilePath
+    )
+
+    if ($PerformPreReqCheck)
+    {
+        try
+        {
+            Write-Host "[Step 1 of 3] Validating and installing the modules required to run the script and validating the user..."
+            Write-Host $([Constants]::SingleDashLine)
+            Write-Host "Setting up prerequisites..."
+            Setup-Prerequisites
+        }
+        catch
+        {
+            Write-Host "Error occurred while setting up prerequisites. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+    }
+    else
+    {
+        Write-Host "[Step 1 of 3] Validating the user..." 
+    }  
+
+    # Connect to Azure account
+    $context = Get-AzContext
+
+    if ([String]::IsNullOrWhiteSpace($context))
+    {
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "Connecting to Azure account..."
+        Connect-AzAccount -Subscription $SubscriptionId -ErrorAction Stop | Out-Null
+        Write-Host "Connected to Azure account." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
```

Comment:
Can we add checks to avoid this call if the context is already set to the intended subscription?